File: Codeforces/C++/problemset/Tram.cpp
// Problem Link: https://codeforces.com/problemset/problem/116/A

#include <iostream>
using namespace std;
typedef long long ll;
int main()
{
    ll n,a,b,c = 0,m = 0;
    cin >> n;
    for(ll i = 0; i < n; i++)
    {
        cin >> a >> b;
        c = (c - a) + b;
        if(c >= m) m = c;
    }
    cout << m;
    return 0;
}

File: Codeforces/Kotlin/src/main/kotlin/problems/Tram.kt
package problems.problems

/* Problem Link: https://codeforces.com/problemset/problem/116/A */

fun main() {
    val n = readln().toInt()
    var current = 0
    var max = 0

    for(i in 1..n) {
        val (a, b) = readln().split(" ").map { it.toInt() }
        current = current - a + b
        if(current > max) max = current
    }

    print(max)
}